src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java

```diff
@@ -101,13 +101,6 @@ public void onServiceFound(NsdServiceInfo service) {
 
       if (stopDiscoveryRunnable.cancel(false)) {
         nsdManager.stopServiceDiscovery(this);
-        if (multicastLock.isHeld()) {
-          multicastLock.release();
-        }
-      }
-
-      if (nsdManagerResolverAvailState != null) {
-        nsdManagerResolverAvailState.acquireResolver();
       }
```

Comment:
@sharadb-amazon 
Right. I moved this code to NsdManagerServiceResolver class.

When I checked the issue, I found that NsdManagerServiceResolver was being called for each instanceName, and when multiple instanceNames were called simultaneously, the discover operation, which operates as the first step in the NsdServiceFinderAndResolver class, had to be run sequentially to find the service. Therefore, this code has been modified to ensure that synchronization is guaranteed for the entire operation, rather than only the resolver operation.

Reply:
But now that would mean, if a service with the instanceName that NsdManagerServiceResolver.resolve was called with, does not respond to the browse request, you'd still end up locking the nsdManagerResolver?

Furthermore, if such a service is never discovered (does not respond to the browse request), we'd keep the nsdManagerResolver locked and not be able to do further resolutions?

Here is the original issue because of which the NsdServiceFinderAndResolver was introduced in the first place: #23709 - can you please take a look and make sure your changes don't regress on that fix?

Reply:
@sharadb-amazon 
Yes, I know why this class was created. Even in the case you mentioned, I understand that the lock can be released when a timeout occurs because there is the code below.

https://github.com/project-chip/connectedhomeip/blob/b742587091630534313e0403d87c649db690496d/src/platform/android/java/chip/platform/NsdManagerServiceResolver.java#L96

Reply:
But that timeout waits for 30 seconds and that will add a noticeable increase in user perceived latency?

Reply:
@sharadb-amazon 
Yes, latency will increase. I think 30 seconds is excessive. I didn't modify this value because I don't know why it was initially set to 30 seconds.

Reply:
@sharadb-amazon maybe we should put the comment here, even make it configurable, I think this is more or less experimental number.

Reply:
Can you add the ability to configure the timeout value? We'd like to mitigate customer impact due to a latency regression.

Reply:
@sharadb-amazon, @yunhanw-google 
Add timeout Parameter in each class.